devices/Mcp7940xx/tests/Mcp7940mTests.cs

```diff
@@ -0,0 +1,1280 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Iot.Device.Common;
+using Iot.Device.Mcp7940xx;
+using nanoFramework.Hardware.Esp32;
+using nanoFramework.TestFramework;
+using System;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Iot.Device.NFUnitTest
+{
+    [TestClass]
+    public class Mcp7940mTests
+    {
+        static I2cDevice _i2cDevice;
+        static Mcp7940m _clock;
+
+        private SpanByte ReadAllBuffers()
+        {
+            // Read all Mcp7940m registers.
+            SpanByte readBuffer = new byte[23];
+
+            _i2cDevice.WriteByte((byte)Register.TimekeepingSecond);
+            _i2cDevice.Read(readBuffer);
+
+            return readBuffer;
+        }
+
+        [Setup]
+        public void SetupMcp7940mTests()
+        {
+            try
+            {
+                Debug.WriteLine("Please adjust for your own usage. If you need another hardware, please add the proper nuget and adjust as well");
+
+                // Setup ESP32 I2C port.
+                Configuration.SetPinFunction(Gpio.IO22, DeviceFunction.I2C1_CLOCK);
+                Configuration.SetPinFunction(Gpio.IO21, DeviceFunction.I2C1_DATA);
+
+                // Setup Mcp7940m device. 
+                I2cConnectionSettings i2cSettings = new I2cConnectionSettings(1, Mcp7940m.DefaultI2cAddress);
+                _i2cDevice = new I2cDevice(i2cSettings);
+                _clock = new Mcp7940m(_i2cDevice, ClockSource.ExternalCrystal);
+
+                // Stop clock updating itself during testing.
+                _clock.Halt();
+
+                // Set clock to default state.
+                _clock.SetTime(new DateTime(2000, 1, 1, 1, 1, 1));
+
+                Mcp7940m.Alarm alarm1 = new Mcp7940m.Alarm(AlarmMatchMode.Full);
+                _clock.SetAlarm1(alarm1);
+
+                Mcp7940m.Alarm alarm2 = new Mcp7940m.Alarm(AlarmMatchMode.Full);
+                _clock.SetAlarm2(alarm2);
+            }
+            catch
+            {
+                Assert.SkipTest("I2C port not supported in this platform or not properly configured");
+            }
+
+        }
+
+        // https://stackoverflow.com/questions/42006662/testing-private-static-generic-methods-in-c-sharp
```

Comment:
Why this comment with the URL here? 🤔 
If it's relevant please add a bit of context. If it's a reminder to yourself, better remove. 😉